modules/acquia_cms_headless/tests/src/Functional/DashboardApiUrlTest.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Drupal\Tests\acquia_cms_headless\Functional;
+
+use Drupal\Tests\acquia_cms_headless\Traits\DashboardTableTrait;
+
+/**
+ * Tests headless dashboard API Url.
+ *
+ * @group acquia_cms
+ * @group acquia_cms_headless
+ * @group medium_risk
+ * @group push
+ */
+class DashboardApiUrlTest extends DashboardWebDriverTestBase {
+
+  use DashboardTableTrait;
+
+  /**
+   * {@inheritdoc}
+   */
+  protected string $sectionTitle = "API URL";
+
+  /**
+   * {@inheritdoc}
+   */
+  protected string $sectionSelector = "#acquia-cms-headless-api-url";
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getHeaders(): array {
+    return [];
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function testButtons(): void {
+    $this->assertButton("Update Base API URL");
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function testSection(): void {
```

Comment:
@deepakmishra2 instead testing the Base API Url in section can test as separate function if possible.
```
/**
   * Tests the base jsonapi URL.
   */
  public function testBaseApiUrl(): void {
    $element = $this->getSection()->find("css", ".headless-dashboard-api-url");
    $this->assertSame($element->getText(), "Base API Url: ". $this->baseUrl . "/jsonapi");
  }
```
  in testSection function you can test image exist or not, any thoughts? 
  cc @vishalkhode1